src/app/component/Slider/Slider.component.js

```diff
@@ -370,9 +349,64 @@ export class Slider extends PureComponent {
         callback.call(this, ...args);
     }
 
-    changeActiveImage(activeImage) {
+    countOffset(active, newActive) {
+        const { children: { length } } = this.props;
+
+        const minIndexOnLeft = active - this.getFocusSlide() + 1;
+        const indexesOnLeft = minIndexOnLeft >= 0
+            ? Array(active - minIndexOnLeft).fill().map((_, idx) => minIndexOnLeft + idx)
+            : Array(active).fill().map((_, idx) => idx).concat(
+                Array(Math.abs(minIndexOnLeft)).fill().map((_, idx) => length + minIndexOnLeft + idx)
+            );
```

Comment:
Such usage of ternary operators is very hard to read. Please refactor and comment on this as well.